src/vs/platform/telemetry/node/workbenchCommonProperties.ts

```diff
@@ -51,3 +53,23 @@ export function resolveWorkbenchCommonProperties(storageService: IStorageService
 // 	storageService.store('telemetry.instanceId', result);
 // 	return result;
 // }
+
+// {{SQL CARBON EDIT}}s
+function setUsageDates(storageService: IStorageService): void {
+	// daily last usage date
+	const dailyLastUseDate = storageService.get('telemetry.dailyLastUseDate') || new Date().toUTCString();
+	storageService.store('telemetry.dailyLastUseDate', dailyLastUseDate);
+
+	// weekly last usage date
+	const weeklyLastUseDate = storageService.get('telemetry.weeklyLastUseDate') || new Date().toUTCString();
+	storageService.store('telemetry.weeklyLastUseDate', weeklyLastUseDate);
+
+	// monthly last usage date
+	const monthlyLastUseDate = storageService.get('telemetry.monthlyLastUseDate') || new Date().toUTCString();
+	storageService.store('telemetry.monthlyLastUseDate', monthlyLastUseDate);
+
+}
+
+export function diffInDays(nowDate: number, lastUseDate: number): number {
```

Comment:
if you define this method in the other file you're editing you would need to import it there.  It doesn't appear to be used in this file.